tests/integration/src/setup.rs

```diff
@@ -79,32 +80,37 @@ mod shibuya {
         }
     }
 
-    /// Deploy a WASM contract with its name. (The code is in `resource/`.)
+    /// Deploy a WASM contract via ALICE as origin. (The code is in `../ink-contracts/`.)
+    /// Assumption: Contract constructor is called "new" and take no arguments
     pub fn deploy_wasm_contract(name: &str) -> AccountId32 {
-        let path = format!("ink-contracts/{}.wasm", name);
-        let code = std::fs::read(path).expect("invalid path");
-        let instantiate_result = Contracts::bare_instantiate(
+        let (address, _) = astar_test_utils::deploy_wasm_contract::<Runtime>(
+            name,
             ALICE,
             0,
             Weight::from_parts(10_000_000_000, 1024 * 1024),
             None,
-            pallet_contracts_primitives::Code::Upload(code),
-            // `new` constructor
             hex::decode("9bae9d5e").expect("invalid data hex"),
-            vec![],
-            pallet_contracts::DebugInfo::Skip,
-            pallet_contracts::CollectEvents::Skip,
         );
 
-        let address = instantiate_result
-            .result
-            .expect("instantiation failed")
-            .account_id;
         // On instantiation, the contract got existential deposit.
         assert_eq!(Balances::free_balance(&address), ExistentialDeposit::get(),);
         address
     }
 
+    /// Call a wasm smart contract method with ALICE as origin
```

Comment:
IMO, a bit too specialized to be under generic utils.

Reply:
I thought so too, but since we already had a similar method `deploy_wasm_contract` with same specialisation (`ALICE` as caller and hardcoded weights), I followed the pattern and put it there.

Reply:
You can always improve the pattern 😁 

The Boy Scout Rule

Reply:
I'll make it less specialised then (add origin as arg), although currently it is only being used by unified accounts CE tests but useful for future CE tests
(IMO, we should have integration tests for all CE, we literally have zero tests for CEs).